### PR TITLE
doc: Document exported functionality of modules

### PR DESCRIPTION
**Pull request by slarse**
This PR adds TSDoc documentation for the exported functionality of all modules. It also adds linting for TSDoc to ensure that the documentation is actually well-formed.

### CODE REVIEW

**Comment by slarse** [comment]
@algomaster99 Actually, wait with a review a second, I see that I've documented promises a bit strangely.

**Comment by slarse** [comment]
Okay, fixed. On the note of promises in returns, should one document a `Promise<void>` return value? I.e. something like `@returns Fulfills with void upon success`? Seems a bit odd, so I didn't, but it _is_ returning something (a promise) and so I wouldn't find it unreasonable to document. Do you have any idea of what is common practice?

File: src/git.ts [hunk 1]
```diff
@@ -3,29 +3,49 @@ import {PathLike} from 'fs';
...
+  /**
+   * @param targetDirectory - Any directory within the worktree of the Git
+   *    repository to work with
+   */
   constructor(targetDirectory: PathLike) {
     this.targetDirectory = targetDirectory;
   }
 
+  /**
+   * Restore all modified files in the current.
```
Comment (algomaster99) [inline src/git.ts:21]: I think you can skip documenting the return type as when we don't define the return type for void functions, I feel a similar logic can be extended here. Moreover, the promise after resolving would anyway return `undefined` so maybe it doesn't make sense.

I am not aware of any convention that instructs one to document `Promise<void>`.

**Review by algomaster99 — APPROVED** [review]
Just dropped one comment. Rest looks good to me :tada:

File: src/git.ts (hunk 1 above)
Comment (slarse) [reply src/git.ts:21]: > I think you can skip documenting the return type as when we don't define the return type for void functions, I feel a similar logic can be extended here. Moreover, the promise after resolving would anyway return undefined so maybe it doesn't make sense.

Right, then I think it makes sense to not add any docs for them, seems rather self-explanatory. Thanks for the input.

**Comment by slarse** [comment]
@algomaster99 Feel free to merge! Just use the PR title as squash message and remove anything from the body :)

**Pull request merged**